From d3ddf88f30fc2844039f670a3e83f989526e674e Mon Sep 17 00:00:00 2001
From: xbw <xbw@rock-chips.com>
Date: Tue, 25 Feb 2014 12:43:01 +0800
Subject: [PATCH] continue with the privous Commit. Complete submission SDMMC
 driver in kernel-3.10.Currently,Only the sdcard can be used.

---
 arch/arm/boot/dts/rk3188-mmc.dtsi  | 2 ++
 drivers/mmc/host/Kconfig           | 2 +-
 drivers/mmc/host/Makefile          | 9 ++++++---
 drivers/mmc/host/dw_mmc-rockchip.c | 4 ++--
 4 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/rk3188-mmc.dtsi b/arch/arm/boot/dts/rk3188-mmc.dtsi
index 2b4eb8d9f207..1305446118b6 100755
--- a/arch/arm/boot/dts/rk3188-mmc.dtsi
+++ b/arch/arm/boot/dts/rk3188-mmc.dtsi
@@ -31,6 +31,8 @@
 	    num-slots = <1>;
 	    supports-highspeed;
 	    broken-cd;
+	    card-detect-delay = <200>;
+	    pwr-gpios = <&gpio3 GPIO_A1 GPIO_ACTIVE_LOW>; /*pwr_en = GPIO3_A1*/
 	    fifo-depth = <0x80>;
 	    mmc,caps = <(MMC_CAP_4_BIT_DATA | MMC_CAP_MMC_HIGHSPEED | MMC_CAP_SD_HIGHSPEED)>;
 	    mmc,ocr =  <(MMC_VDD_25_26 | MMC_VDD_26_27 | MMC_VDD_27_28 | MMC_VDD_28_29 |
diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index 0fc7a9eed0be..cae551376801 100755
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -547,7 +547,7 @@ config MMC_DW_PLTFM
 
 	  If unsure, say Y.
 
-config MMC_DW_RK
+config MMC_DW_ROCKCHIP
 	tristate "Rockchip specific extensions for Synopsys DW Memory Card Interface"
 	depends on MMC_DW
 	select MMC_DW_PLTFM
diff --git a/drivers/mmc/host/Makefile b/drivers/mmc/host/Makefile
index 81cd30728cb5..9ce558427521 100755
--- a/drivers/mmc/host/Makefile
+++ b/drivers/mmc/host/Makefile
@@ -39,11 +39,14 @@ obj-$(CONFIG_MMC_SDHI)		+= sh_mobile_sdhi.o
 obj-$(CONFIG_MMC_CB710)		+= cb710-mmc.o
 obj-$(CONFIG_MMC_VIA_SDMMC)	+= via-sdmmc.o
 obj-$(CONFIG_SDH_BFIN)		+= bfin_sdh.o
-obj-$(CONFIG_MMC_DW)		+= dw_mmc.o
+
+#obj-$(CONFIG_MMC_DW)		+= dw_mmc.o
+obj-$(CONFIG_MMC_DW)           += rk_sdmmc.o
+
 obj-$(CONFIG_MMC_DW_PLTFM)	+= dw_mmc-pltfm.o
 
-# added by XBW to Specific Extensions for Synopsys DW Multimedia Card Interface in Rockchip Soc.
-obj-$(CONFIG_MMC_DW_RK)         += dw_mmc-rockchip.o rk_sdmmc_of.o
+# To Specific Extensions for Synopsys DW Multimedia Card Interface in Rockchip Soc. Added by XBW.
+obj-$(CONFIG_MMC_DW_ROCKCHIP)         += dw_mmc-rockchip.o rk_sdmmc_of.o
 
 obj-$(CONFIG_MMC_DW_EXYNOS)	+= dw_mmc-exynos.o
 obj-$(CONFIG_MMC_DW_PCI)	+= dw_mmc-pci.o
diff --git a/drivers/mmc/host/dw_mmc-rockchip.c b/drivers/mmc/host/dw_mmc-rockchip.c
index 86c12333dba0..0d07c55bdd32 100755
--- a/drivers/mmc/host/dw_mmc-rockchip.c
+++ b/drivers/mmc/host/dw_mmc-rockchip.c
@@ -20,7 +20,7 @@
 #include <linux/of.h>
 #include <linux/of_gpio.h>
 
-#include "dw_mmc.h"
+#include "rk_sdmmc.h"
 #include "dw_mmc-pltfm.h"
 
 #include "rk_sdmmc_of.h"
@@ -223,6 +223,6 @@ static struct platform_driver dw_mci_rockchip_pltfm_driver = {
 module_platform_driver(dw_mci_rockchip_pltfm_driver);
 
 MODULE_DESCRIPTION("Rockchip Specific DW-SDMMC Driver Extension");
-MODULE_AUTHOR("Bangwang Xie < xbw@rock-chips.com");
+MODULE_AUTHOR("Bangwang Xie < xbw@rock-chips.com>");
 MODULE_LICENSE("GPL v2");
 MODULE_ALIAS("platform:dwmmc-rockchip");
-- 
2.35.3

